compiler/parser/src/string_parser.rs

```diff
@@ -219,35 +219,19 @@ impl<'a> StringParser<'a> {
                         Some('a') => ConversionFlag::Ascii,
                         Some('r') => ConversionFlag::Repr,
                         Some(_) => {
-                            return Err(if expression.trim().is_empty() {
-                                EmptyExpression.to_lexical_error(self.get_pos())
-                            } else {
-                                InvalidConversionFlag.to_lexical_error(self.get_pos())
-                            });
+                            return Err(InvalidConversionFlag.to_lexical_error(self.get_pos()));
                         }
                         None => {
-                            return Err(if expression.trim().is_empty() {
-                                EmptyExpression.to_lexical_error(self.get_pos())
-                            } else {
-                                UnclosedLbrace.to_lexical_error(self.get_pos())
-                            });
+                            return Err(UnclosedLbrace.to_lexical_error(self.get_pos()));
                         }
                     };
 
-                    if let Some(&peek) = self.peek() {
-                        if peek != '}' && peek != ':' {
-                            return Err(if expression.trim().is_empty() {
-                                EmptyExpression.to_lexical_error(self.get_pos())
-                            } else {
-                                UnclosedLbrace.to_lexical_error(self.get_pos())
-                            });
+                    match self.peek() {
+                        Some('}' | ':') => {}
+                        Some(_) => {
+                            return Err(InvalidConversionFlag.to_lexical_error(self.get_pos()))
```

Comment:
seems like this should still complain about the unclosed brace, i.e `f"{20!ao}"` results in `SyntaxError: f-string: expecting '}'`

Reply:
Looks like I copy & pasted a wrong error. I'll fix this.

Reply:
Fixed.